compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -625,23 +625,35 @@ int32_t OMR::X86::CodeGenerator::getMaximumNumbersOfAssignableGPRs()
    }
 
 /*
- * this method returns TRUE for all the cases we decide NOT to inline the CAS native
- * it checks the same condition as inlineCompareAndSwapNative in tr.source/trj9/x/codegen/J9TreeEvaluator.cpp
+ * This method returns TRUE for all the cases we decide NOT to replace the call to CAS
```

Comment:
Instead of tinkering with this function in OMR and adding even more Java-specific code, I would rather see it moved directly into the OpenJ9 project.  It is a static helper function that is ONLY called from code guarded with `#ifdef J9_PROJECT_SPECIFIC`.  You should be able to make it extern, hoist it out of this file with relatively little effort, and then do whatever is necessary for Java.

The comment below also suggests that there's similar logic in J9TreeEvaluator that it must be kept in sync with.  It seems to me that having both of those checks in the same project would be a step toward commoning them into one and would be easier to maintain.

Reply:
I'm very much in favour of this, but would it be ok if I opened a new issue for the work? I'd like to get the Unsafe fix into the code promptly, and this would be my first attempt to move code between projects, so I don't want to try and rush it.

Reply:
I think that moving it down is a good idea, but since it is broken at present in at least some cases fixing it then moving it seems reasonable to me - that way we only move working code which is probably a good principle to have in general when there is a choice. Your thoughts @0xdaryl ?

Reply:
OK, I'm fine with an issue.

Reply:
Opened #1942